src/wfs/src/main/java/org/geoserver/wfs/xml/FeatureTypeSchemaBuilder.java

```diff
@@ -780,8 +780,14 @@ private XSDComplexTypeDefinition buildComplexSchemaContent(ComplexType complexTy
                         Class binding = attribute.getType().getBinding();
                         typeName = findTypeName(binding);
                         if (typeName == null) {
-                            throw new NullPointerException("Could not find a type for property: "
-                                    + attribute.getName() + " of type: " + binding.getName());
+                            // Fallback on String
+                            logger.finer("Fallback mapping: attribute " + attribute.getName() + " to String" );
```

Comment:
Guard logging :-)

Reply:
Yep. At the very beginning I was thinking to add guard but I have noticed that the class has several other logs without any guard so I didn't introduce that there.